### DISCUSSION TITLE
Deployment pattern in the AWS ECS environment

### DISCUSSION THREAD

**Discussion by huynhphat22** [open-telemetry/opentelemetry-collector-contrib#35792] [Q&A]
Hi,

We are currently doing our POC with Opentelemetry in AWS ECS environment with Datadog as the backend monitoring provider. Our services are running on **Fargate Windows**. We would love know:

1. Is any deployment pattern where we don't have to use OTEL agents within the containers to collect host metrics/logs?
2. We did try with OTEL agents pattern but we are currently having issue with inaccurate/missing metrics (We're using awsecscontainermetrics receiver). Is this a recognized/common issue and is there a way to fix this?

**Note:** We are not using ADOT collector as it does not work with Datadog (Confirmed by AWS tech support). 

Our Implementation is below. 
![image](https://github.com/user-attachments/assets/ad98d446-b081-43b6-9e32-6088acda3ec2)

**Comment by Yogagh** [comment]
Hello,
Can someone respond to our question as it is very important for us to know whether we need to include the OTEL collector as part of the app image (in App ECS TaskDefinition) in-order to collect the ECS Task and Container based logs & metrics?
